packages/backend/src/peripherals/database/RpcTransactionCountRepository.ts

```diff
@@ -51,42 +52,55 @@ export class RpcTransactionCountRepository extends BaseRepository {
   async getMissingRangesByProject(projectId: ProjectId) {
     const knex = await this.knex()
 
-    const noNext = (await knex.raw(
+    const blockNumbers = (await knex.raw(
       `
       WITH 
         project_blocks AS (
           SELECT * FROM transactions.rpc WHERE project_id=?
-        )
-      SELECT * 
-      FROM (
-        SELECT project_blocks.block_number 
-        FROM project_blocks 
-        LEFT JOIN project_blocks b2 ON project_blocks.block_number  = b2.block_number - 1
-        WHERE b2.block_number IS NULL) AS no_next
-      ORDER BY block_number ASC
-    `,
-      projectId.toString(),
-    )) as unknown as RawBlockNumberQueryResult
-
-    const noPrev = (await knex.raw(
-      `
-      WITH 
-          project_blocks AS (
-            SELECT * FROM transactions.rpc WHERE project_id=?
-          )
-        SELECT * 
-        FROM (
-          SELECT project_blocks.block_number 
+        ),
+        no_next AS (
```

Comment:
I think it **might be possible** that `no_next` and `no_prev` being part of `WITH` statement would slow things down - the alternative would be to keep only `project_blocks` inside `WITH`.